### ISSUE TITLE
iRule stats does not have all available events

### ISSUE THREAD

**Issue by walterk82** [F5Networks/f5-telemetry-streaming#86]
### Environment
 * Telemetry Streaming Version: 1.15.0
 * BIG-IP Version: 14.1.2.7

### Summary
When using the default OOTB System Poller, the iRules stats are incomplete.  When there is an iRule that has multiple events to report, TS only sends the _last_ event to an external system.

### Steps To Reproduce
1. Submit the following declaration:
```json
{
    "class": "Telemetry",
    "schemaVersion": "1.15.0",
    "Controls": {
        "class": "Controls",
        "logLevel": "info",
        "debug": false
    },
    "Memory_Endpoints": {
        "class": "Telemetry_Endpoints",
        "items": {
            "memory": {
                "path": "/mgmt/tm/sys/memory/stats"
            }
        }
    },
    "SystemData_Poller": {
        "class": "Telemetry_System_Poller",
        "interval": 60,
        "enable": true
    },
    "System": {
        "class": "Telemetry_System",
        "systemPoller": [
            "SystemData_Poller"
        ],
        "enable": true
    },
    "Default_Pull_Consumer": {
        "class": "Telemetry_Pull_Consumer",
        "systemPoller": [
            "SystemData_Poller"
        ],
        "enable": true,
        "trace": false
    },
    "Splunk_Consumer": {
        "class": "Telemetry_Consumer",
        "type": "Splunk",
        "protocol": "https",
        "host": "splunk-hec",
        "path": "/services/collector/event",
        "port": 443,
        "passphrase": {
            "cipherText": "<token>",
            "class": "Secret",
            "protected": "SecureVault"
        },
        "enable": true,
        "trace": false
    }
}
```

2. Observe the following error response:
There is no error on the declaration itself.

### Expected Behavior
While I don't have a specific expected result, here is the direct output from native REST.
```json
{
    "kind": "tm:ltm:rule:rulestats",
    "generation": 1,
    "selfLink": "https://localhost/mgmt/tm/ltm/rule/~QAS~Entry-Access-Rule/stats?ver=14.1.2.7",
    "entries": {
        "https://localhost/mgmt/tm/ltm/rule/~QAS~Entry-Access-Rule/~QAS~Entry-Access-Rule:ACCESS_SAML_ASSERTION/stats": {
            "nestedStats": {
                "kind": "tm:ltm:rule:rulestats",
                "selfLink": "https://localhost/mgmt/tm/ltm/rule/~QAS~Entry-Access-Rule/~QAS~Entry-Access-Rule:ACCESS_SAML_ASSERTION/stats?ver=14.1.2.7",
                "entries": {
                    "aborts": {
                        "value": 0
                    },
                    "avgCycles": {
                        "value": 11759
                    },
                    "eventType": {
                        "description": "ACCESS_SAML_ASSERTION"
                    },
                    "failures": {
                        "value": 0
                    },
                    "maxCycles": {
                        "value": 18627
                    },
                    "minCycles": {
                        "value": 8139
                    },
                    "tmName": {
                        "description": "/QAS/Entry-Access-Rule"
                    },
                    "priority": {
                        "value": 500
                    },
                    "totalExecutions": {
                        "value": 129
                    }
                }
            }
        },
        "https://localhost/mgmt/tm/ltm/rule/~QAS~Entry-Access-Rule/~QAS~Entry-Access-Rule:ACCESS_SAML_SLO_REQ/stats": {
            "nestedStats": {
                "kind": "tm:ltm:rule:rulestats",
                "selfLink": "https://localhost/mgmt/tm/ltm/rule/~QAS~Entry-Access-Rule/~QAS~Entry-Access-Rule:ACCESS_SAML_SLO_REQ/stats?ver=14.1.2.7",
                "entries": {
                    "aborts": {
                        "value": 0
                    },
                    "avgCycles": {
                        "value": 0
                    },
                    "eventType": {
                        "description": "ACCESS_SAML_SLO_REQ"
                    },
                    "failures": {
                        "value": 0
                    },
                    "maxCycles": {
                        "value": 0
                    },
                    "minCycles": {
                        "value": 0
                    },
                    "tmName": {
                        "description": "/QAS/Entry-Access-Rule"
                    },
                    "priority": {
                        "value": 500
                    },
                    "totalExecutions": {
                        "value": 0
                    }
                }
            }
        },
        "https://localhost/mgmt/tm/ltm/rule/~QAS~Entry-Access-Rule/~QAS~Entry-Access-Rule:ACCESS_SAML_SLO_RESP/stats": {
            "nestedStats": {
                "kind": "tm:ltm:rule:rulestats",
                "selfLink": "https://localhost/mgmt/tm/ltm/rule/~QAS~Entry-Access-Rule/~QAS~Entry-Access-Rule:ACCESS_SAML_SLO_RESP/stats?ver=14.1.2.7",
                "entries": {
                    "aborts": {
                        "value": 0
                    },
                    "avgCycles": {
                        "value": 0
                    },
                    "eventType": {
                        "description": "ACCESS_SAML_SLO_RESP"
                    },
                    "failures": {
                        "value": 0
                    },
                    "maxCycles": {
                        "value": 0
                    },
                    "minCycles": {
                        "value": 0
                    },
                    "tmName": {
                        "description": "/QAS/Entry-Access-Rule"
                    },
                    "priority": {
                        "value": 500
                    },
                    "totalExecutions": {
                        "value": 0
                    }
                }
            }
        },
        "https://localhost/mgmt/tm/ltm/rule/~QAS~Entry-Access-Rule/~QAS~Entry-Access-Rule:ACCESS_SESSION_STARTED/stats": {
            "nestedStats": {
                "kind": "tm:ltm:rule:rulestats",
                "selfLink": "https://localhost/mgmt/tm/ltm/rule/~QAS~Entry-Access-Rule/~QAS~Entry-Access-Rule:ACCESS_SESSION_STARTED/stats?ver=14.1.2.7",
                "entries": {
                    "aborts": {
                        "value": 0
                    },
                    "avgCycles": {
                        "value": 241963
                    },
                    "eventType": {
                        "description": "ACCESS_SESSION_STARTED"
                    },
                    "failures": {
                        "value": 0
                    },
                    "maxCycles": {
                        "value": 4070145
                    },
                    "minCycles": {
                        "value": 92271
                    },
                    "tmName": {
                        "description": "/QAS/Entry-Access-Rule"
                    },
                    "priority": {
                        "value": 500
                    },
                    "totalExecutions": {
                        "value": 46489
                    }
                }
            }
        },
        "https://localhost/mgmt/tm/ltm/rule/~QAS~Entry-Access-Rule/~QAS~Entry-Access-Rule:HTTP_REQUEST/stats": {
            "nestedStats": {
                "kind": "tm:ltm:rule:rulestats",
                "selfLink": "https://localhost/mgmt/tm/ltm/rule/~QAS~Entry-Access-Rule/~QAS~Entry-Access-Rule:HTTP_REQUEST/stats?ver=14.1.2.7",
                "entries": {
                    "aborts": {
                        "value": 0
                    },
                    "avgCycles": {
                        "value": 156982
                    },
                    "eventType": {
                        "description": "HTTP_REQUEST"
                    },
                    "failures": {
                        "value": 0
                    },
                    "maxCycles": {
                        "value": 14689017
                    },
                    "minCycles": {
                        "value": 36861
                    },
                    "tmName": {
                        "description": "/QAS/Entry-Access-Rule"
                    },
                    "priority": {
                        "value": 600
                    },
                    "totalExecutions": {
                        "value": 1743962
                    }
                }
            }
        },
        "https://localhost/mgmt/tm/ltm/rule/~QAS~Entry-Access-Rule/~QAS~Entry-Access-Rule:HTTP_RESPONSE/stats": {
            "nestedStats": {
                "kind": "tm:ltm:rule:rulestats",
                "selfLink": "https://localhost/mgmt/tm/ltm/rule/~QAS~Entry-Access-Rule/~QAS~Entry-Access-Rule:HTTP_RESPONSE/stats?ver=14.1.2.7",
                "entries": {
                    "aborts": {
                        "value": 0
                    },
                    "avgCycles": {
                        "value": 20687
                    },
                    "eventType": {
                        "description": "HTTP_RESPONSE"
                    },
                    "failures": {
                        "value": 0
                    },
                    "maxCycles": {
                        "value": 4613973
                    },
                    "minCycles": {
                        "value": 6996
                    },
                    "tmName": {
                        "description": "/QAS/Entry-Access-Rule"
                    },
                    "priority": {
                        "value": 500
                    },
                    "totalExecutions": {
                        "value": 1560136
                    }
                }
            }
        },
        "https://localhost/mgmt/tm/ltm/rule/~QAS~Entry-Access-Rule/~QAS~Entry-Access-Rule:HTTP_RESPONSE_RELEASE/stats": {
            "nestedStats": {
                "kind": "tm:ltm:rule:rulestats",
                "selfLink": "https://localhost/mgmt/tm/ltm/rule/~QAS~Entry-Access-Rule/~QAS~Entry-Access-Rule:HTTP_RESPONSE_RELEASE/stats?ver=14.1.2.7",
                "entries": {
                    "aborts": {
                        "value": 0
                    },
                    "avgCycles": {
                        "value": 68222
                    },
                    "eventType": {
                        "description": "HTTP_RESPONSE_RELEASE"
                    },
                    "failures": {
                        "value": 0
                    },
                    "maxCycles": {
                        "value": 4930527
                    },
                    "minCycles": {
                        "value": 5109
                    },
                    "tmName": {
                        "description": "/QAS/Entry-Access-Rule"
                    },
                    "priority": {
                        "value": 500
                    },
                    "totalExecutions": {
                        "value": 1671228
                    }
                }
            }
        }
    }
}
```

### Actual Behavior
```json
/QAS/Entry-Access-Rule": {
    "events": {
        "HTTP_RESPONSE_RELEASE": {
            "aborts": 0,
            "avgCycles": 66229,
            "failures": 0,
            "maxCycles": 4930527,
            "minCycles": 5109,
            "priority": 500,
            "totalExecutions": 1668951
        }
    },
    "name": "/QAS/Entry-Access-Rule",
    "tenant": "QAS"
}
```

**Comment by walterk82** [comment]
Cross referenced with F5 Support case C3406892

**Comment by walterk82** [comment]
custom endpoints does return all of the raw data
```json
    "Custom_Endpoints": {
        "class": "Telemetry_Endpoints",
        "basePath": "/mgmt/tm/",
        "items": {
            "iRules": {
                "path": "ltm/rule/stats",
                "enable": true
            }
        },
        "enable": true
    },
```

**Comment by dstokesf5** [comment]
Thank you for reporting this. I have added it to our internal product backlog as AUTOTOOL-1967. I will be keeping an eye on the support case to make sure everyone stays on the same page. Thank you for providing the case ID.

**Comment by walterk82** [comment]
assigned  Bug ID 958397

**Comment by walterk82** [comment]
Is there an estimate of when this bug will be addressed?  When using the custom_endpoint mode, the JSON output structure is significantly different and will cause significant migration challenges for Splunk parsing to accommodate both models.  I need to understand where to place our focus on processing this data.  Thank-You.

**Comment by dstokesf5** [comment]
The bug fix will be available in the next release, which should be available within a week.

**Comment by walterk82** [comment]
Verified fixed with 1.16
